### PR TITLE
fix(artwork): never serve artist folder images as album art

### LINKED ISSUE #5599: [Bug]: Album art replaced with artist artwork on 0.62.0, doesn't happen on 0.61.0
### I confirm that:

- [x] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [x] I'm using the latest version (your issue may have been fixed already)

### Version

0.62.0

### Current Behavior

Album artwork shows the artist image, not the embedded album art in the grid / library. Rolling back to 0.61.0 fixes the problem. The album art in the circular player shows the album art. The log shows this: evel=warning msg="Parent
folder not found for album cover art lookup". Oddly it only impacts about 30% of albums in my library. Almost all of my album art is embedded.

### Expected Behavior

The album art shows on the albums. This is what happens on 0.61.0.

### Steps To Reproduce

Pull 0.62.0 docker image and log in.

### Environment

```markdown
- OS: Docker / NixOS
- Browser: Firefox
- Client: Web client and Arpeggi on iOS
```

### How Navidrome is installed?

Docker

### Configuration

```toml
services:
  navidrome:
    image: deluan/navidrome:0.61.0
    container_name: navidrome
    user: 1000:100 # UID:GID should be owner of volumes
#    ports:
#      - "4533:4533"
    restart: unless-stopped
    env_file:
      - .env
    environment:
      # Optional: put your config options customization here. Exampl>
      ND_LASTFM_ENABLED: true
      ND_LASTFM_APIKEY: ${ND_LASTFM_APIKEY}
      ND_LASTFM_SECRET: ${ND_LASTFM_SECRET}
      ND_SPOTIFY_ID: ${ND_SPOTIFY_ID}
      ND_SPOTIFY_SECRET: ${ND_SPOTIFY_SECRET}
      ND_ENABLEINSIGHTSCOLLECTOR: false
    volumes:
      - "/docker/navidrome:/data"
      - "/music/music:/music:ro"
    networks:
      - proxy
networks:
  proxy:
    external: true
```

### Relevant log output

```shell
time="2026-06-11T23:54:50Z" level=warning msg="Parent
folder not found for album cover art lookup"
parentID=4MRzhRRPJn1ly0X8oZaEHH
requestId=2d89ab2669d9/WsGbYtYdNn-000047

time="2026-06-11T23:55:43Z" level=warning msg="Parent
folder not found for album cover art lookup"
parentID=4MRzhRRPJn1ly0X8oZaEHH
requestId=2d89ab2669d9/WsGbYtYdNn-000092

time="2026-06-11T23:55:43Z" level=warning msg="Parent
folder not found for album cover art lookup"
parentID=4MRzhRRPJn1ly0X8oZaEHH
requestId=2d89ab2669d9/WsGbYtYdNn-000091
```

### Anything else?

_No response_

### Code of Conduct

- [x] I agree to follow Navidrome's Code of Conduct

### PR DESCRIPTION
**Pull request by deluan**
### Description
Since v0.62.0, the album cover-art parent-folder fallback (#5451/#5457) could pick up images from the **artist** folder: any album without image files in its own folder(s) inherited the artist thumbnail (e.g. `Artist/folder.jpg`) as its cover art. Single-disc albums spread across sibling folders were also affected — there the artist image could even shadow the album's own art, due to `CoverArtPriority` pattern order.

The fix gates the parent-folder inclusion with a structural check instead of folder-name matching: the common parent only qualifies as an album root when no audio belonging to other albums lives anywhere under it. An artist folder contains other albums' tracks, while an album root above disc subfolders contains only the album's own, so this works for any disc-subfolder naming scheme and keeps the multi-disc behavior from #5376/#5456 intact. The check is exposed as a new `FolderRepository.HasAudioOutsideFolders` method, and the parent resolution was extracted into a dedicated helper.

Known residual: an artist whose *only* album has no images anywhere is structurally indistinguishable from an album root, so that one layout still inherits the artist image.

### Related Issues
Fixes #5599 

Regression introduced by #5451/#5457 (related to #5376 and #5456). Reported [on Reddit](https://www.reddit.com/r/navidrome/comments/1u06x0g/comment/oqlz5ep/).

### Type of Change
- [x] Bug fix
- [ ] New feature
- [ ] Documentation update
- [x] Refactor
- [ ] Other (please describe):

### Checklist
Please review and check all that apply:

- [x] My code follows the project’s coding style
- [x] I have tested the changes locally
- [ ] I have added or updated documentation as needed
- [x] I have added tests that prove my fix/feature works (or explain why not)
- [x] All existing and new tests pass

### How to Test
1. Create a library like:
   ```
   Artist/folder.jpg            ← artist thumbnail
   Artist/Album A/01.mp3        (no images)
   Artist/Album B/01.mp3
   Artist/Album B/cover.jpg
   ```
2. Scan and open Album A. Before this PR it shows the artist thumbnail; with it, the placeholder.
3. Album B, multi-disc layouts (`Album/CD1`, `CD2` + root `cover.jpg`), and single-disc-subfolder layouts keep their current artwork.

The new e2e specs cover these scenarios: `make test PKG=./core/artwork/e2e`.

### Additional Notes
The artwork resolution code is accumulating real complexity by trying to handle ever more layout edge cases — multi-disc roots, disc subfolders with arbitrary names, spread albums, parent fallbacks, and now structural artist-folder detection. This may be a sign we should rethink the artwork implementation (e.g. classify folders at scan time instead of guessing at read time), and possibly drop support for some of these edge cases in the future.

### CODE REVIEW

File: persistence/folder_repository.go [hunk 1]
```diff
@@ -188,6 +189,35 @@ func (r folderRepository) queryFolderUpdateInfo(where And) (map[string]model.Fol
...
+	query := r.newSelect().Columns("count(*)").Where(And{
+		Eq{"library_id": parent.LibraryID, "missing": false},
+		Gt{"num_audio_files": 0},
+		NotEq{"id": excludeFolderIDs},
+		Or{
+			// Direct children have path = parentPath; deeper descendants match the prefix
+			Eq{"path": parentPath},
+			Expr(`path LIKE ? ESCAPE '\'`, escapeLikePrefix(parentPath)+"/%"),
+		},
+	})
```
[the remark this answers is not in the document]
Comment (deluan) [reply persistence/folder_repository.go]: squirrel never generates `NOT IN ()`: an empty slice in `NotEq` compiles to the `(1=1)` constant (see `expr.go` in squirrel v1.5.4). `excludeFolderIDs` is also never empty here — albums always have at least one folder. Keeping the unconditional clause.

File: core/artwork/reader_album.go [hunk 2]
```diff
@@ -159,6 +143,50 @@ func loadAlbumFoldersPaths(ctx context.Context, ds model.DataStore, albums ...mo
...
+	}
+	parent, err := ds.Folder(ctx).Get(commonParentID)
+	if errors.Is(err, model.ErrNotFound) {
+		log.Warn(ctx, "Parent folder not found for album cover art lookup", "parentID", commonParentID)
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	if parent.ParentID == "" {
```
[the remark this answers is not in the document]
Comment (deluan) [reply core/artwork/reader_album.go:176]: After the refactor, `ErrNotFound` returns early, so `parent` is only dereferenced when `Get` returned no error — the repository contract never yields `(nil, nil)`. A nil guard here would be dead code protecting against a broken mock contract.

File: persistence/folder_repository_test.go [hunk 3]
```diff
@@ -217,6 +217,67 @@ var _ = Describe("FolderRepository", func() {
...
+			Expect(repo.HasAudioOutsideFolders(*albumRoot, []string{disc1.ID, disc2.ID})).To(BeTrue())
+		})
+
+		It("ignores audio outside the parent's subtree", func() {
+			other := model.NewFolder(testLib, "TestHasAudio/Other Album")
+			other.NumAudioFiles = 10
+			Expect(repo.Put(other)).To(Succeed())
+
+			Expect(repo.HasAudioOutsideFolders(*albumRoot, []string{disc1.ID, disc2.ID})).To(BeFalse())
+		})
```
[the remark this answers is not in the document]
Comment (deluan) [reply persistence/folder_repository_test.go:261]: Gomega's `Expect(actual, extra...)` is variadic: the extra `error` return is accepted and asserted to be nil. These tests compile and pass — the suite has run green on this branch. No change needed.

File: persistence/folder_repository.go [hunk 4]
```diff
@@ -188,6 +189,35 @@ func (r folderRepository) queryFolderUpdateInfo(where And) (map[string]model.Fol
...
+	count, err := r.count(query)
+	return count > 0, err
```
[the remark this answers is not in the document]
Comment (deluan) [reply persistence/folder_repository.go]: Switched to the repository's canonical `exists` helper in 664a733d2. Note it is itself `count(*)`-based, and the result is cached per artwork key downstream, so the practical impact is simplification rather than performance.
